overlord/snapstate/policy/os.go

```diff
@@ -44,7 +46,10 @@ func (p *osPolicy) CanRemove(st *state.State, snapst *snapstate.SnapState, rev s
 		return nil
 	}
 
-	if p.modelBase == "" {
+	// if the base is unset and dev.IsCoreBoot is true, then we know this is a
+	// UC16 system. note that base might be unset on classic models, so we must
+	// check dev.IsCoreBoot as well.
```

Comment:
```suggestion
	// consider the case of a UC16 system, where the model does not specify a base, 
	// since 'core' is already implied and is actively used by the system, 
	// which boots in the UC way
```